.github/workflows/ttnn-run-sweeps.yaml

```diff
@@ -2,40 +2,92 @@ name: "ttnn - Run sweeps"
 
 on:
   workflow_dispatch:
+  schedule:
+    - cron: "0 21 * * *" # This cron schedule runs the workflow at 8:00pm nightly
+  push:
+    branches:
+      - jdesousa/sweeps-ci-schedule
 
 jobs:
   build-artifact:
     uses: ./.github/workflows/build-artifact.yaml
     secrets: inherit
 
-  ttnn-sweeps:
+  ttnn-generate-sweeps:
     needs: build-artifact
+    env:
+      TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
+      ARCH_NAME: wormhole_b0
+      ELASTIC_USERNAME: ${{ secrets.SWEEPS_ELASTIC_USERNAME }}
+      ELASTIC_PASSWORD: ${{ secrets.SWEEPS_ELASTIC_PASSWORD }}
+      LD_LIBRARY_PATH: ${{ github.workspace }}/build/lib
+    environment: dev
+    timeout-minutes: 30
+    runs-on: [build, in-service]
+    steps:
+      - uses: tenstorrent-metal/metal-workflows/.github/actions/checkout-with-submodule-lfs@v2.0.0
+      - name: Set up dynamic env vars for build
+        run: |
+          echo "TT_METAL_HOME=$(pwd)" >> $GITHUB_ENV
+      - uses: ./.github/actions/prepare-metal-run
+        with:
+          arch: wormhole_b0
+      - name: Run ttnn sweeps generation
+        run: |
+          source ${{ github.workspace }}/python_env/bin/activate
+          cd $TT_METAL_HOME
+          export PYTHONPATH=$TT_METAL_HOME
+          python tests/sweep_framework/parameter_generator.py --elastic cloud --tag ci-main --explicit
+
+  ttnn-run-sweeps:
+    needs: ttnn-generate-sweeps
     strategy:
       # Do not fail-fast because we need to ensure all tests go to completion
       # so we try not to get hanging machines
       fail-fast: false
       matrix:
-        arch: [grayskull, wormhole_b0]
+        test-group:
+          [
+            {
+              name: "Grayskull E150 Sweeps",
+              arch: grayskull,
+              runs-on: ["cloud-virtual-machine", "E150", "in-service"],
+              tt-smi-cmd: "tt-smi-metal -r 0"
+            },
+            {
+              name: "Wormhole N150 Sweeps",
+              arch: wormhole_b0,
+              runs-on: ["cloud-virtual-machine", "N150", "in-service"],
+              tt-smi-cmd: "tt-smi-metal -r 0"
+            },
+            {
+              name: "Wormhole N300 Sweeps",
+              arch: wormhole_b0,
+              runs-on: ["cloud-virtual-machine", "N300", "in-service"],
+              tt-smi-cmd: "tt-smi-metal -r 0"
+            }
+          ]
     env:
       TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
-      ARCH_NAME: ${{ matrix.arch }}
+      ARCH_NAME: ${{ matrix.test-group.arch }}
       ELASTIC_USERNAME: ${{ secrets.SWEEPS_ELASTIC_USERNAME }}
       ELASTIC_PASSWORD: ${{ secrets.SWEEPS_ELASTIC_PASSWORD }}
+      TT_SMI_RESET_COMMAND: ${{ matrix.test-group.tt-smi-cmd }}
       LD_LIBRARY_PATH: ${{ github.workspace }}/build/lib
     environment: dev
     timeout-minutes: 720
```

Comment:
Does this timeout need to be this long?
Not sure if tests need to be 12 hours long.

Reply:
I did a run of it last night, and even at 12 hours it timed out eventually. There is a lot of tests, and a lot of them have hangs which require a reset which takes a long time. I was talking with @xanderchin and @tt-rkim we will need to do some sort of parallelization of the tests across runners. I thought 12 hours might be enough for now but even already it's too short.
https://github.com/tenstorrent/tt-metal/actions/runs/10603188781

Reply:
So what is the next steps for this PR?
Is there anymore work to be done? ie parallelization of runs?
It doesn't look like tests will pass in its current state?

Reply:
Most of the time being taken right now is because @sjameelTT has added some large suites for concat that all hang. (#12073) He is going to add some asserts to catch a lot of these hanging cases which should shorten this significantly (without these tests it takes 3-4 hours rn). Parallelization is going to come after, but I think we can merge this to actually have the pipeline working but maybe time out for the time being. Not sure if you want to wait for #12073 to be completed or not before we merge this?